src/lucky/tags/text_helpers.cr

```diff
@@ -0,0 +1,233 @@
+module Lucky::TextHelpers
+  @@_cycles = Hash(String, Cycle).new
+
+  def truncate(text : String, length : Int32 = 30, omission : String = "...", separator : String | Nil = nil, escape : Bool = false, blk : Nil | Proc = nil)
+    if text
+      content = truncate_text(text, length, omission, separator)
+      content = escape ? HTML.escape(content) : content
+      content += blk.call.to_s if blk.is_a?(Proc) && text.size > length
```

Comment:
It's more common to check that something is not `nil` like this `if !blk.nil?`, rather than checking it's type.

Reply:
Interesting. This works in this scenario where blk is either Nil or Proc.

Side note, I struggle with the optional Block/Proc a bit with this in Crystal. The Ruby way might be to check if the object respond_to? but the Crystal convention seems to force the object to conform to a type or interface.